### PR TITLE
Docs and refactoring

### PR DESCRIPTION
**Pull request by lintusj1**
Added inline documentation to `methods.py` and did some light refactoring to `Rejection`.

### CODE REVIEW

File: elfi/methods/methods.py [hunk 1]
```diff
@@ -111,40 +256,56 @@ def infer(self, *args, **kwargs):
...
-    def to_array(self, batch, outputs=None):
+    def _to_array(self, batch, outputs=None):
         """Helper method to turn batches into numpy array
         
         Parameters
         ----------
         batch : dict or list
            Batch or list of batches
-        outputs : list, optional
+        outputs : list
```
Comment (vuolleko) [inline elfi/methods/methods.py]: Isn't this still optional?

Comment (lintusj1) [inline elfi/methods/methods.py]: Yes, I'll fix

File: elfi/methods/methods.py [hunk 2]
```diff
@@ -39,15 +109,25 @@ class InferenceMethod(object):
...
+            in every batch.
         batch_size : int
         seed : int
+            Seed for the data generation from the ElfiModel
+        pool : OutputPool
+            OutputPool both stores and provides precomputed values for batches.
+        max_parallel_batches : int
+            Maximum number of batches allowed to be in computation at the same time.
+            Defaults to number of cores in the client + 1
+
```
Comment (lintusj1) [inline elfi/methods/methods.py]: Will change to number of cores

File: elfi/methods/methods.py [hunk 3]
```diff
@@ -85,24 +167,87 @@ def parameters(self):
...
+        Returns
+        -------
+        None
+        """
         raise NotImplementedError
 
     @property
-    def estimated_total_batches(self):
-        """Can change during execution if needed"""
+    def _n_total_batches(self):
```
Comment (lintusj1) [inline elfi/methods/methods.py]: I'll change it to a method.

**Pull request merged**